### ISSUE TITLE
No margin when using ImageIcon and text together, left justified, in JButton

### ISSUE THREAD

**Issue by eclab** [JFormDesigner/FlatLaf#746]
Recent MacOS.  If I do 

    button = new JButton("whatever");  // I can't add the image in the constructor in my case
    button.setHorizontalAlignment(SwingConstants.LEFT);
    button.setText(text);
    button.setIcon(bigIcon);    // 32x32 say, but it doesn't really matter

... I get button with the icon smashed up against the left side with zero margin.  I cannot find any properties that change matters.  setMargin() is ignored.  The only option I have is to literally change the image to add some N pixels of transparent zero-padding.  It looks awful so given how nice the kit looks in general, I can only assume that it's not intentional.

**Comment by DevCharly** [comment]
Please post a screenshot.

FlatLaf version?
Java version?

**Comment by eclab** [comment]
<img width="528" alt="Screen Shot 2023-10-13 at 11 08 58 AM" src="https://github.com/JFormDesigner/FlatLaf/assets/12685780/ddf9abb1-52d1-4a1c-bf99-bb8f583821f2">

In this image, there are three buttons: the two in the list at left, and one in the main window.  As can be seen, the image is smooshed up against the left side with no margin, and no option to add a margin.

MacOS 12.6 Monterey, M1, openjdk version "20.0.2" 2023-07-18, flatlaf-3.1.1.jar

**Comment by DevCharly** [comment]
Works fine for me:

![button-left-align](https://github.com/JFormDesigner/FlatLaf/assets/5604048/d15b4c0e-be63-43a6-9f7b-5e204a8318b2)

~~~java
public class ButtonTest
{
	public static void main( String[] args ) {
		SwingUtilities.invokeLater( () -> {
			FlatLightLaf.setup();

			JFrame frame = new JFrame( "test" );
			frame.setDefaultCloseOperation( JFrame.EXIT_ON_CLOSE );


			JButton button1 = new JButton( "Button", UIManager.getIcon( "OptionPane.errorIcon" ) );
			button1.setHorizontalAlignment( SwingConstants.LEFT );

			JButton button2 = new JButton( "Button with insets 30, 50, 10, 10", UIManager.getIcon( "OptionPane.errorIcon" ) );
			button2.setHorizontalAlignment( SwingConstants.LEFT );
			button2.setMargin( new Insets( 30, 50, 10, 10 ) );

			JPanel panel = new JPanel( new BorderLayout() );
			panel.setBorder( new EmptyBorder( 20, 20, 20, 20 ) );
			panel.add( button1, BorderLayout.NORTH );
			panel.add( button2, BorderLayout.SOUTH );

			frame.add( panel );
			frame.setSize( 500, 200 );
			frame.setVisible( true );
		} );
	}
}
~~~

**Comment by eclab** [comment]
Your example fails when the text is HTML (which I need).  Try changing the text

    Button with insets 30, 50, 10, 10

to

    <html>Button with insets 30, 50, 10, 10</html>

Sean

**Comment by DevCharly** [comment]
Screenshot from previous post:

![Screen Shot 2023-10-13 at 12 48 55 PM](https://github.com/JFormDesigner/FlatLaf/assets/12685780/19ed17c6-01e2-460f-a50b-3042b06b7ea4)

**Comment by DevCharly** [comment]
Thanks, with HTML text I can reproduce the problem.

Seems to be a **bug in Java 19 and later**.
Works fine with Java 18 and older.

Metal, Nimbus and Windows L&F have the same problem:

![image](https://github.com/JFormDesigner/FlatLaf/assets/5604048/6203632b-a519-45be-b37d-d36c47ae9df1)

Will check what changed in JDK 19 and whether it is possible to workaround/fix the issue...

**Comment by DevCharly** [comment]
The "fix" for JDK bug https://bugs.openjdk.org/browse/JDK-8282772 causes the problem (see PR https://github.com/openjdk/jdk/pull/8407)

From my view, JDK-8282772 is an invalid report...

**Comment by DevCharly** [comment]
I've reported the bug to https://bugreport.java.com/bugreport/

I've also posted some info to the JDK PR. See https://github.com/openjdk/jdk/pull/8407#issuecomment-1761583430 and https://github.com/openjdk/jdk/pull/8407#issuecomment-1762926635.

Screenshot that demonstrates the problem (shows Metal L&F):

![HTML-buttons](https://github.com/JFormDesigner/FlatLaf/assets/5604048/9b1e2779-a893-4860-9990-e3876683f0e8)

**Comment by DevCharly** [comment]
I've implemented a workaround for FlatLaf, which is now available in 3.2.2

**Comment by DevCharly** [comment]
related JDK bug: https://bugs.openjdk.org/browse/JDK-8318590
